File: output/copilot/java/Solution03_cp.java
public class Solution03_cp {
    public int strStr(String haystack, String needle) {
        if (needle.length() == 0) {
            return 0;
        }
        if (haystack.length() == 0) {
            return -1;
        }
        int i = 0;
        int j = 0;
        while (i < haystack.length()) {
            if (haystack.charAt(i) == needle.charAt(j)) {
                if (j == needle.length() - 1) {
                    return i - j;
                }
                j++;
            } else {
                i -= j;
                j = 0;
            }
            i++;
        }
        return -1;
    }
}

File: output/copilot/js/Solution03_cp.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 */
var strStr = function(haystack, needle) {
    if (needle.length === 0) return 0;
    for (let i = 0; i < haystack.length; i++) {
        if (haystack[i] === needle[0]) {
            let j = 0;
            while (j < needle.length && haystack[i + j] === needle[j]) {
                j++;
            }
            if (j === needle.length) return i;
        }
    }
    return -1;    
};